src/shared/components/plots/MultipleCategoryBarPlotUtils.ts

```diff
@@ -98,6 +98,71 @@ export function sortDataByCategory<D>(
         }
     });
 }
+export function getSortedMajorCategories(
+    data: IMultipleCategoryBarPlotData[],
+    sortByOption: string | undefined
+): string[] {
+    if (sortByOption === 'SortByTotalSum') {
+        const majorCategoryCounts: any = {};
+
+        data.forEach(item => {
```

Comment:
i think you can do this operation with flatMap, which will avoid nesting forEach (which is confusing)

Reply:
The above function is used to find the sum of each `majorCategory` for different `minorCategories`. For instance, consider the following example:

data contains three `minorCategory` names: Biopsy, Resection, and Cusa. Each of these `minorCategories` contains a large array named counts, and each element in that array is an object containing `majorCategory`, count, and percentage. For example, for Biopsy, the `majorCategories` might include Germ Cell Tumor and Thyroid Cancer....

I am maintaining a totalSum of these majorCategories and then applying sorting.

If I use flatMap, it would first be difficult to determine which majorCategory belongs to which minorCategory. Additionally, I think it would require more space and have an equivalent time complexity of O(n * m) as it does now with the two nested loops. First, I would need to perform the flatMap operation (a map operation followed by a flattening operation, essentially equivalent to a nested loop) and then use reduce to accumulate the counts. What are your suggestions should i follow another approach?